gradio/helpers.py

```diff
@@ -599,7 +542,7 @@ def load_from_cache(self, example_id: int) -> list[Any]:
             example_id: The id of the example to process (zero-indexed).
         """
         if self.cache_examples == "lazy":
-            if cached_index := self._get_cached_index_if_cached(example_id) is None:
+            if (cached_index := self._get_cached_index_if_cached(example_id)) is None:
```

Comment:
This was the underlying bug. The walrus operator has a [lower order of precedence](https://peps.python.org/pep-0572/#relative-precedence-of) than every other operator except the comma, so in this case `cached_index` was being assigned a boolean value (`False`) in the `else` branch, which was being implicitly converted to a `0`, which would then lead to the first cached example always being selected.

Reply:
dammit python

Reply:
Wow